File: question/leetcode/leetcode0014.Longest-Common-Prefix/leetcode0014.go
package leetcode

func longestCommonPrefix(strs []string) string {
	n := len(strs)
	if n == 0 {
		return ""
	}
	for i := 0; i < len(strs[0]); i++ {
		for j := 1; j < len(strs); j++ {
			if i == len(strs[j]) || strs[j][i] != strs[0][i] {
				return strs[0][:i]
			}
		}
	}
	return strs[0]
}
func longestCommonPrefixII(strs []string) string {
	n := len(strs)
	if n == 0 {
		return ""
	}
	prefix := ""
	firstStr := strs[0]
	for i := 0; i < len(firstStr); i++ {
		char := firstStr[i]
		for j := 1; j < len(strs); j++ {
			//防止索引越界
			if i >= len(strs[j]) || strs[j][i] != char {
				return prefix
			}
		}
		prefix += string(char)
	}

	return prefix
}

File: question/leetcode/leetcode0014.Longest-Common-Prefix/leetcode0014.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        int n = strs.length;
        if (n == 0) {
            return "";
        }
        String prefix = "";
        String firstStr = strs[0];
        for (int i = 0; i < firstStr.length(); i++) {
            char ch = firstStr.charAt(i);
            for (int j = 1; j < strs.length; j++) {
                if (i >= strs[j].length() || strs[j].charAt(i) != ch) {
                    return prefix;
                }
            }
            prefix += ch;
        }

        return prefix;
    }

}
